cmd/clusterctl/client/yamlprocessor/processor.go

```diff
@@ -0,0 +1,35 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package yamlprocessor
+
+import "sigs.k8s.io/cluster-api/cmd/clusterctl/client/config"
+
+// Processor defines the methods necessary for creating a specific yaml
+// processor.
+type Processor interface {
+
+	// ArtifactName returns the name of the template artifacts that need to be
+	// retrieved from the source.
+	ArtifactName(version, flavor string) string
```

Comment:
+1, also maybe prefix with `Get` to be consistent?

Reply:
Yup...I see what you are saying. 
Since this method is technically returning the name of the "cluster" template to be used it would make sense to not be generic. I think I was using artifact because I was thinking about tar.gz blobs and didn't think of them as the actual templates.

I'll update this. Thanks!